src/mgr/PyOSDMap.cc

```diff
@@ -189,19 +191,24 @@ static int
 BasePyOSDMap_init(BasePyOSDMap *self, PyObject *args, PyObject *kwds)
 {
     PyObject *osdmap_capsule = nullptr;
-    static const char *kwlist[] = {"osdmap_capsule", NULL};
+    PyObject *tp_capsule = nullptr;
+    static const char *kwlist[] = {"osdmap_capsule", "tp_capsule", NULL};
 
-    if (! PyArg_ParseTupleAndKeywords(args, kwds, "O",
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "OT",
                                       const_cast<char**>(kwlist),
-                                      &osdmap_capsule)) {
+                                      &osdmap_capsule,
+                                      &tp_capsule)) {
```

Comment:
I'm not sure why the thread should be attached to the py osdmap object.  It seems like a general purpose worker thread should be a global property of the mgr, not tied to this particular object?

Reply:
you are right.  it's simply because I don't know how to pass the global thread pool to the specific py caller, like balancer...